Sources/Core/Views/GenericButton/GenericButton.swift

```diff
@@ -0,0 +1,176 @@
+//
+//  GenericButton.swift
+//  PovioKit
+//
+//  Created by Arlind Dushi on 3/17/22.
+//  Copyright © 2022 Povio Inc. All rights reserved.
+//
+
+import SwiftUI
+
+@available(iOS 13, *)
+public class GenericButtonProperties: ObservableObject {
+  @Published public var title: String = ""
+  @Published public var font: Font = .system(size: 18)
+  @Published public var textColor: Color = .white
+  @Published public var cornerRadius: GenericButton.CornerRadiusType = .custom(0)
+  @Published public var backgroundType = GenericButton.Background.plain(.blue)
+  @Published public var borderColor: Color = .clear
+  @Published public var borderWidth: CGFloat = 0
+  @Published public var extraImage: GenericButton.ExtraImage?
+}
+
+@available(iOS 13, *)
+public struct GenericButton: View {
```

Comment:
To be honest, no offense, this name sucks 🤷‍♂️ I have a few alternatives in mind
- CTA (aka Call To Action)
- ButtonControl
- ActionButton

@poviolabs/ios-developers what do you think. Please vote.

Reply:
_ActionButton_ sounds good to me